### ISSUE TITLE
set priority for layers?

### ISSUE THREAD

**Issue by dieterdreist** [mapbox/mapbox-gl-js#9843]
**mapbox-gl-js version**:1.10.1

### Question
I would like to set the rendering priority of different layers / overlays. For example I would prefer to have more priority for the overlay than for the basemap. And more priority for small overlays than for large ones. Basically I would like to set priorities for different elements according to my needs. Is it possible?
(I have the overlay data locally, but still sometimes the map coming from the web is rendered before the overlay).

**Comment by mourner** [comment]
No, rendering priorities are fully defined by the order of layers and features within each layer. There's no concept of a "z-index" in GL JS.

**Comment by dieterdreist** [comment]
when I wrote priority I did not mean rendering order, I meant the process of (automatic) simplification. Geometries are reduced in detail by mapbox-gl-js when zoomed out, and when zooming in they remain simplified for some seconds, until the screen is completely recreated. I have several of these overlays and would like to influence which is “de-simplified” first.

Similarly, I have the user position leave blurred traces behind, and sometimes these a lagging (the userposition marker changes position several times without leaving traces, until the renderer keeps up and all the traces show up at once).
